lib/infrastructure/models/chat_model.dart

```diff
@@ -11,12 +11,12 @@ class ChatModel {
   const ChatModel({required this.ts, required this.message, required this.writerUid, required this.writerName});
 
   factory ChatModel.fromMap(Map<String, dynamic> json) {
-    final ts = json['ts'].toDate();
```

Comment:
why does this not work? fix the error there ~ maybe ChatModel should set ts as servertimestamp (if possible)...so ts is always a TimeStamp on the db, and then this line always works

Reply:
I have made an extension for parse any  datetime object (String, TimeStamp, Int, etc) to DateTime .